src/java.base/share/classes/java/lang/invoke/StringConcatFactory.java

```diff
@@ -489,17 +491,22 @@ private static MethodHandle generateMHInlineCopy(MethodType mt, String[] constan
                 // and prependers for char, boolean
                 ptypes[i] = int.class;
             } else if (cl == Object.class) {
-                filter = objectStringifier();
+                if (objFilters == null) {
+                    objFilters = new MethodHandle[ptypes.length];
+                }
+                objFilters[i] = objectStringifier();
+                ptypes[i] = String.class;
             } else if (cl == float.class) {
-                filter = floatStringifier();
+                if (floatFilters == null) {
+                    floatFilters = new MethodHandle[ptypes.length];
+                }
+                floatFilters[i] = floatStringifier();
+                ptypes[i] = String.class;
             } else if (cl == double.class) {
-                filter = doubleStringifier();
-            }
-            if (filter != null) {
-                if (filters == null) {
-                    filters = new MethodHandle[ptypes.length];
+                if (doubleFilters == null) {
+                    doubleFilters = new MethodHandle[ptypes.length];
                 }
-                filters[i] = filter;
+                doubleFilters[i] = doubleStringifier();
```

Comment:
Why do we have `ptypes[i] = int.class` for `int` and `short`, `ptypes[i] = String.class` for `float`, `double` and `Object` and none for `boolean` and `char`? 🧐